src/validation.cpp

```diff
@@ -5001,11 +5001,24 @@ bool ChainstateManager::PopulateAndValidateSnapshot(
         // Fake nChainTx so that GuessVerificationProgress reports accurately
         index->nChainTx = index->pprev ? index->pprev->nChainTx + index->nTx : 1;
 
+        if (!index->IsValid(BLOCK_VALID_SCRIPTS)) {
+            // This flag will be removed once the block is fully validated by a
+            // background chainstate.
+            index->nStatus |= BLOCK_ASSUMED_VALID;
+        }
+
```

Comment:
In commit "validation: set BLOCK_ASSUMED_VALID during snapshot load" (9a8ad497238c8c8aba1ba8679bc7cdbe409c635e)

Similar to comment in last commit it would seem nice to just be able to drop this interaction/toggling of flags and just do unconditional `index->nStatus |= BLOCK_ASSUMED_VALID`

Reply:
I don't think it makes sense to apply the ASSUMED_VALID classification for blocks we've already validated, especially since (i) it's easy to avoid, and (ii) there will almost certainly be blocks at the base of the chain that have been validated prior to loading a UTXO snapshot.